### PR TITLE
Better travis config

### PR DESCRIPTION
**Pull request by Nyholm**
I also cleaned the code a bit

FYI @bocharsky-bw, I'll merge this and tag 0.4.0

### CODE REVIEW

File: src/FileStorage.php [hunk 1]
```diff
@@ -49,28 +47,26 @@
...
-     * @param SymfonyTranslationLoader|TranslationLoader|TranslationReader $loader
-     * @param array                                                        $dir
-     * @param array                                                        $options
+     * @param TranslationWriterInterface $writer
+     * @param TranslationReaderInterface $reader
+     * @param array                      $dir
+     * @param array                      $options
      */
-    public function __construct(TranslationWriter $writer, $loader, array $dir, array $options = [])
+    public function __construct($writer, $reader, array $dir, array $options = [])
```
Comment (bocharsky-bw) [inline src/FileStorage.php:60]: I think we can still keep `TranslationWriter` typehint for `$writer`, it's the same class where just a method was renamed.

File: src/FileStorage.php [hunk 2]
```diff
@@ -49,28 +47,26 @@
...
-        // Create a legacy loader which is a wrapper for TranslationReader
-        if ($loader instanceof TranslationReader) {
-            $loader = new LegacyTranslationLoader($loader);
-        }
-        if (!$loader instanceof SymfonyTranslationLoader && !$loader instanceof TranslationLoader) {
-            throw new \LogicException('Second parameter of FileStorage must be a Symfony translation loader or implement Translation\SymfonyStorage\TranslationLoader');
+
+        // Create a wrapper for legacy reader
+        if (!$reader instanceof TranslationReaderInterface) {
+            $reader = new LegacyTranslationReader($reader);
```
Comment (bocharsky-bw) [inline src/FileStorage.php:69]: Nice idea to invert my logic with `LegacyTranslationReader` 👍 

I somehow didn't notice this way

File: src/LegacyTranslationWriter.php [hunk 3]
```diff
@@ -15,20 +15,22 @@
...
-class LegacyTranslationWriter
+final class LegacyTranslationWriter // implements Symfony\Component\Translation\Writer\TranslationWriterInterface
 {
-    /**
-     * @var TranslationWriter
-     */
     private $writer;
 
-    public function __construct(TranslationWriter $writer)
+    public function __construct($writer)
```
Comment (bocharsky-bw) [inline src/LegacyTranslationWriter.php:27]: The same here, I think we can still use `TranslationWriter` type hint. Or do you want to show a better-explained error with throwing `LogicException` in this constructor?

**Review by bocharsky-bw — COMMENTED** [review]
Nice changes 👍

File: src/FileStorage.php (hunk 1 above)
Comment (Nyholm) [reply src/FileStorage.php:60]: Yeah, but we want people to be able to provide their own class. 

If the BC layer was removed I would type hint this to TranslationWriterInterface.

File: src/LegacyTranslationWriter.php (hunk 3 above)
Comment (Nyholm) [reply src/LegacyTranslationWriter.php:27]: Exactly. I removed it here just because I wanted a little better exception message.

**Pull request merged**